packages/cardano-graphql/src/DataProjection/blockHandlers/AssetBlockHandler.ts

```diff
@@ -1,9 +1,33 @@
-import { QueryResult, QueryVariables, RollBackwardContext, RollForwardContext } from '../types';
+import {
+  BlockHandler,
+  CombinedProcessingResult,
+  ProcessParameters,
+  ProcessingResult,
+  QueryResult,
+  QueryVariables,
+  RollBackwardContext,
+  RollForwardContext
+} from '../types';
+import { MetadataClient } from '../MetadataClient';
 import { Schema, isAlonzoBlock, isMaryBlock } from '@cardano-ogmios/client';
 import { dummyLogger } from 'ts-log';
 
-export const createAssetBlockHandler = (logger = dummyLogger) => ({
-  id: 'Asset',
+const HANDLER_ID = 'Asset';
+
+export const createAssetBlockHandler = (metadataClient: MetadataClient, logger = dummyLogger): BlockHandler => ({
+  id: HANDLER_ID,
+  process: async ({ queryResult }: ProcessParameters) => {
+    const assetsToInsert = [];
+    if (queryResult) {
+      for (const asset of queryResult.data.assets) {
+        logger.info('About to fetch asset metadata');
+        if (!(await metadataClient.fetch([asset.assetId]))) {
```

Comment:
I will check with @rhyslbw, maybe I misunderstood it.
I've noticed that it can be more performant if I only do one query. The problem I see is how can the asset be mapped to the corresponding metadata? It looks like the response does not contain that information. Also, how would be the insert?  Looks like graphql's asset metadata schema and the response differs in some fields, should I map them?

Reply:
> The problem I see is how can the asset be mapped to the corresponding metadata

I would guess it corresponds to index, e.g. `fetch(['id1', 'id2'])` => `[asset1metadata, asset2metadata]`

> Also, how would be the insert? Looks like graphql's asset metadata schema and the response differs in some fields, should I map them?

Does the metadata not cover graphql schema? Please try to map it and leave comments on any missing data.